src/osd/PG.cc

```diff
@@ -2669,9 +2669,15 @@ std::pair<ghobject_t, bool> PG::do_delete_work(
       &olist,
       &next);
     if (!olist.empty()) {
-      dout(0) << __func__ << " additional unexpected onode list"
-              <<" (new onodes has appeared since PG removal started"
-              << olist << dendl;
+      for (auto& oid : olist) {
+        if (oid == pgmeta_oid) {
+          dout(20) << __func__ << " removing pgmeta object " << oid << dendl;
```

Comment:
I think I'm missing something - where are we removing the pgmeta object?

Reply:
this is done by PGLog::clear_info_log() below

Reply:
It happens as a part of the final cleanup https://github.com/ceph/ceph/blob/master/src/osd/PG.cc#L2708-L2713, verified it in the logs

```
2021-04-22T21:42:28.416+0000 7fa19827e700 20 osd.0 pg_epoch: 16 pg[2.0( v 15'758 (0'0,15'758] lb MIN local-lis/les=13/14 n=758 ec=13/13 lis/c=13/13 les/c/f=14/14/0 sis=16 pruub=14.636423111s) [] r=-1 lpr=16 DELETING pi=[13,16)/1 crt=15'758 lcod 15'757 mlcod 0'0 unknown NOTIFY pruub 114.589828491s@ mbc={}] do_delete_work removing pgmeta object #2:00000000::::head#
2021-04-22T21:42:28.416+0000 7fa19827e700 10 bluestore(/home/nojha/work/ceph/build/dev/osd0) queue_transactions ch 0x565375807840 2.0_head
2021-04-22T21:42:28.416+0000 7fa19827e700 20 bluestore(/home/nojha/work/ceph/build/dev/osd0) _txc_create osr 0x56537577d4a0 = 0x56537574f480 seq 791
2021-04-22T21:42:28.416+0000 7fa19827e700 20 _pin0x565367d68000   #2:00000000::::head# pinned
2021-04-22T21:42:28.416+0000 7fa19827e700 15 bluestore(/home/nojha/work/ceph/build/dev/osd0) _remove 2.0_head #2:00000000::::head# onode 0x565375901180 txc 0x56537574f480
2021-04-22T21:42:28.416+0000 7fa19827e700 15 bluestore(/home/nojha/work/ceph/build/dev/osd0) _do_truncate 2.0_head #2:00000000::::head# 0x0
2021-04-22T21:42:28.416+0000 7fa19827e700 20 bluestore.onode(0x565375901180).flush flush done
2021-04-22T21:42:28.420+0000 7fa19827e700 20 bluestore(/home/nojha/work/ceph/build/dev/osd0) _do_omap_clear remove range start: 0x000000000000049F2D end: 0x000000000000049F7E
2021-04-22T21:42:28.420+0000 7fa19827e700 10 bluestore(/home/nojha/work/ceph/build/dev/osd0) _remove 2.0_head #2:00000000::::head# = 0
2021-04-22T21:42:28.420+0000 7fa19827e700 15 bluestore(/home/nojha/work/ceph/build/dev/osd0) _remove_collection 2.0_head
2021-04-22T21:42:28.420+0000 7fa19827e700 20 bluestore.OnodeSpace(0x5653758079f8 in 0x565367d68000) map_any
2021-04-22T21:42:28.420+0000 7fa19827e700 20 bluestore(/home/nojha/work/ceph/build/dev/osd0) _collection_list range #-4:00000000::::0#0 to #-4:ffffffff::::0#0 and #2:00000000::::0#0 to #2:ffffffff::::0#0 start GHMIN
2021-04-22T21:42:28.420+0000 7fa19827e700 20 bluestore(/home/nojha/work/ceph/build/dev/osd0) _collection_list pend #-4:ffffffff::::0#0
2021-04-22T21:42:28.420+0000 7fa19827e700 20 bluestore(/home/nojha/work/ceph/build/dev/osd0) _collection_list oid #-1:30f0e826:::osdmap.13:0# >= #-4:ffffffff::::0#0
2021-04-22T21:42:28.424+0000 7fa19827e700 20 bluestore(/home/nojha/work/ceph/build/dev/osd0) _collection_list oid #2:00000000::::head# end GHMAX
2021-04-22T21:42:28.424+0000 7fa19827e700 20 bluestore(/home/nojha/work/ceph/build/dev/osd0) _collection_list iterator not valid (end of db?)
2021-04-22T21:42:28.424+0000 7fa19827e700 10 bluestore(/home/nojha/work/ceph/build/dev/osd0) _remove_collection oid #2:00000000::::head#
```